File: 763Partition.py
class Solution(object):
    def partitionLabels(self, S):
        dic = {c: i for i, c in enumerate(S)}
        arr = []
        left = 0
        right = 0
        print(dic)
        for i, c in enumerate(S):
            right = max(right, dic[c])
            if(right == i):
                arr.append(right-left+1)
                left = right+1
        return arr


a = Solution()
print(a.partitionLabels("ababcbacadefegdehijhklij"))

File: 763Partition.js
const { set } = require("lodash");

var partitionLabels = function (S) {
  let set = {};
  for (var i = 0; i < S.length; i++) {
    set[S.charAt(i)] = i;
  }
  let left = 0;
  let right = 0;
  let res = [];
  for (var i = 0; i < S.length; i++) {
    right = Math.max(right, set[S[i]]);
    if (i == right) {
      res.push(right - left + 1);
      left = right + 1;
    }
  }
  return res;
};

console.log(partitionLabels("ababcbacadefegdehijhklij"));
